dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BitstreamRestRepository.java

```diff
@@ -135,6 +147,69 @@ protected void delete(Context context, UUID id) throws AuthorizeException {
         }
     }
 
+    /**
+     * Find the bitstream for the provided handle and sequence or filename.
+     * When a bitstream can be found with the sequence ID it will be returned if the user has "METADATA_READ" access.
+     *
+     * @param handle    The handle of the item
+     * @param sequence  The sequence ID of the bitstream
+     * @param filename  The filename of the bitstream
+     *
+     * @return a Page of BitstreamRest instance matching the user query
+     */
+    @SearchRestMethod(name = "byItemHandle")
+    public BitstreamRest findByItemHandle(@Parameter(value = "handle", required = true) String handle,
+                                          @Parameter(value = "sequence") Integer sequence,
+                                          @Parameter(value = "filename") String filename) {
+        if (StringUtils.isBlank(filename) && sequence == null) {
+            throw new IllegalArgumentException("The request should include a sequence or a filename");
+        }
+
+        try {
+            Context context = obtainContext();
+            DSpaceObject dSpaceObject = handleService.resolveToObject(context, handle);
+
+            if (!(dSpaceObject instanceof Item)) {
+                throw new UnprocessableEntityException("The provided handle does not correspond to an existing item");
+            }
+            Item item = (Item) dSpaceObject;
+
+            List<Bitstream> matchedBitstreams = getMatchedBitstreams(item, sequence, filename);
+
+            if (matchedBitstreams.isEmpty()) {
+                return null;
+            } else {
+                return converter.toRest(matchedBitstreams.get(0), utils.obtainProjection());
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    private List<Bitstream> getMatchedBitstreams(Item item, Integer sequence, String filename) {
```

Comment:
@MarieVerdonck : now that you removed the permission check, this method can likely be simplified to return a single Bitstream (i.e. the first that matches).  The logic is essentially the same (first check for a sequence match, then a filename match), but you don't need to gather Bitstreams into a List anymore.  Just return the first match, or `null` if none match.